src/core/agent/services/identifierService.ts

```diff
@@ -140,35 +150,116 @@ class IdentifierService extends AgentService {
     };
   }
 
+  async resolvePendingIdentifier() {
+    const pendingIdentifierCreation = await this.basicStorage.findById(
+      MiscRecordId.IDENTIFIERS_PENDING_CREATION
+    );
+
+    if (
+      pendingIdentifierCreation &&
+      Array.isArray(pendingIdentifierCreation?.content?.name)
+    ) {
+      for (const name of pendingIdentifierCreation.content.name) {
+        let metadata: Omit<IdentifierMetadataRecordProps, "id" | "createdAt">;
+        const [themeString, rest] = name.split(":");
+        if (!rest) {
+          throw new Error(
+            `${IdentifierService.IDENTIFIER_METADATA_RECORD_MISSING}`
+          );
+        }
+
+        const theme = Number(themeString);
+        const groupMatch = rest.match(/^(\d)-(.+)-(.+)$/);
+        if (groupMatch) {
+          const [, initiatorFlag, groupId, displayName] = groupMatch;
+          metadata = {
+            theme,
+            displayName,
+            groupMetadata: {
+              groupId,
+              groupInitiator: initiatorFlag === "1",
+              groupCreated: false,
+            },
+          };
+        } else {
+          metadata = {
+            theme,
+            displayName: rest,
+          };
+        }
+
+        await this.createIdentifier(metadata, true);
+      }
+    }
+  }
+
   @OnlineOnly
   async createIdentifier(
-    metadata: Omit<IdentifierMetadataRecordProps, "id" | "createdAt">
-  ): Promise<CreateIdentifierResult> {
-    const startTime = Date.now();
-
+    metadata: Omit<IdentifierMetadataRecordProps, "id" | "createdAt">,
+    isResolvePendingCration?: boolean
+  ): Promise<void> {
     if (!identifierTypeThemes.includes(metadata.theme)) {
       throw new Error(`${IdentifierService.THEME_WAS_NOT_VALID}`);
     }
 
-    const witnesses = await this.getAvailableWitnesses();
-    if (witnesses.length === 0) {
-      throw new Error(IdentifierService.NO_WITNESSES_AVAILABLE);
-    }
-
     let name = `${metadata.theme}:${metadata.displayName}`;
     if (metadata.groupMetadata) {
       const initiatorFlag = metadata.groupMetadata.groupInitiator ? "1" : "0";
       name = `${metadata.theme}:${initiatorFlag}-${metadata.groupMetadata.groupId}:${metadata.displayName}`;
     }
+    let namesPendingCreation = [name];
+
+    const existRecord = await this.basicStorage.findById(
+      MiscRecordId.IDENTIFIERS_PENDING_CREATION
+    );
+
+    if (
+      existRecord &&
+      Array.isArray(existRecord?.content?.name) &&
+      !isResolvePendingCration
+    ) {
+      const isNameExist = existRecord.content.name.includes(name);
+      namesPendingCreation = [...existRecord.content.name, name];
+
+      if (isNameExist) {
+        throw new Error(`${IdentifierService.IDENTIFIER_NAME_TAKEN}: ${name}`);
+      }
+    }
+
+    await this.props.signifyClient
```

Comment:
We no longer need this call now since the UI will hit it further down (a bunch of these comments don't really apply any more since our discussion on slack :/)